### DISCUSSION TITLE
Passing different authorization token to different virtual users in artillery api test in yml file.

### DISCUSSION THREAD

**Discussion by BhavnaSoni1991** [artilleryio/artillery#2770] [Artillery]
I am writing performance tests for api using artillery and for this I am only maintaining scenario.yml file and no js file. I am passing authorization token through Authorization header in yml file. How can I pass different authorization header for each virtual user?

**Comment by hassy** [comment]
It depends on how those authorization tokens are generated. One option is to use a CSV file with a number of pre-defined tokens and use `config.payload` to make them available to the VUs in the test. The main caveat here is that you won't be able to guarantee that those tokens are not reused. Another option is to use a custom JS hook and generate/fetch those tokens from somewhere else before the VU starts (via a `beforeScenario` hook - https://www.artillery.io/docs/reference/engines/http#function-actions-and-beforescenario--afterscenario-hooks)
